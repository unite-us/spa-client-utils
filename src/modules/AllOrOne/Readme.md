### Require Confirmation

```jsx
initialState = {
  showAll: false,
};

const DummyLocation = ({ location }) => (
  <div className="detail-card-text__location mb-quarter">
    <Icon className="detail-card-text__icon" icon="IconMapMarker" size={20} />
    <p title={location.displayName}>
      {location.displayName} ({location.distance})
    </p>
  </div>
);

const locations = [
  {
    displayName: "123 lol Street, Funmouth, MA 12345",
    distance: "11.41 mi"
  },
  {
    displayName: "123 Loooooooooooooooooooong Street, Funmouth, MA 12345",
    distance: "40.29 mi"
  }
];

<div>
  <button
    onClick={() => setState({ showAll: !state.showAll })}
    style={{ marginBottom: '30px' }}
  >
    {state.showAll ? 'See Less' : 'See More'}
  </button>
  
  <div>
    <h3>Locations</h3>
    <AllOrOne showAll={state.showAll}>
      {
        locations.map((location, index) => <DummyLocation location={location} key={index} />)
      }
    </AllOrOne>
  </div>
</div>
```
