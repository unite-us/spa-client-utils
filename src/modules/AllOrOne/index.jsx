import React from 'react';
import PropTypes from 'prop-types';

const AllOrOne = (props) => {
  if (props.showAll) {
    return React.Children.toArray(props.children);
  }

  const renderedChildren = React.Children.toArray(props.children);
  return renderedChildren.length > 0 ? renderedChildren[0] : null;
};

AllOrOne.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.node,
  ]),
  showAll: PropTypes.bool,
};

AllOrOne.defaultProps = {
  showAll: false,
};

export default AllOrOne;
