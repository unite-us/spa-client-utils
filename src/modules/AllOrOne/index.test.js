import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import AllOrOne from './index';

describe('AllOrOne', () => {
  const DummyLocation = ({ location }) => (
    <div>
      <p title={location.displayName}>
        {location.displayName} ({location.distance})
      </p>
    </div>
  );

  DummyLocation.propTypes = {
    location: PropTypes.object.isRequired,
  };

  const locations = [
    {
      displayName: '123 lol Street, Funmouth, MA 12345',
      distance: '11.41 mi',
    },
    {
      displayName: '123 Street, Funmouth, MA 12345',
      distance: '40.29 mi',
    },
  ];

  describe('returns null', () => {
    test('if the collection is empty', () => {
      expect(renderer.create(
        <AllOrOne>
          {
            [].map(location => (
              <DummyLocation key={location.displayName} location={location} />
            ))
          }
        </AllOrOne>,
      )).toMatchSnapshot();
    });

    test('if the collection is empty and showAll is true', () => {
      expect(renderer.create(
        <AllOrOne showAll>
          {
            [].map(location => (
              <DummyLocation key={location.displayName} location={location} />
            ))
          }
        </AllOrOne>,
      )).toMatchSnapshot();
    });

    test('if nothing is passed in', () => {
      expect(renderer.create(
        <AllOrOne />,
      )).toMatchSnapshot();
    });

    test('nothing is passed in and showAll is true', () => {
      expect(renderer.create(
        <AllOrOne showAll />,
      )).toMatchSnapshot();
    });
  });

  describe('returns the first in a collection', () => {
    test('if showAll is false', () => {
      expect(renderer.create(
        <AllOrOne>
          {
            locations.map(location => (
              <DummyLocation key={location.displayName} location={location} />
            ))
          }
        </AllOrOne>,
      )).toMatchSnapshot();
    });
  });

  describe('returns all the collection', () => {
    test('if showAll is true', () => {
      expect(renderer.create(
        <AllOrOne showAll>
          {
            locations.map(location => (
              <DummyLocation key={location.displayName} location={location} />
            ))
          }
        </AllOrOne>,
      )).toMatchSnapshot();
    });
  });
});
