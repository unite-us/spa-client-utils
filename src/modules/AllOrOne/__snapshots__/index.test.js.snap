// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOrOne returns all the collection if showAll is true 1`] = `
Array [
  <div>
    <p
      title="123 lol Street, Funmouth, MA 12345"
    >
      123 lol Street, Funmouth, MA 12345
       (
      11.41 mi
      )
    </p>
  </div>,
  <div>
    <p
      title="123 Street, Funmouth, MA 12345"
    >
      123 Street, Funmouth, MA 12345
       (
      40.29 mi
      )
    </p>
  </div>,
]
`;

exports[`AllOrOne returns null if nothing is passed in 1`] = `null`;

exports[`AllOrOne returns null if the collection is empty 1`] = `null`;

exports[`AllOrOne returns null if the collection is empty and showAll is true 1`] = `null`;

exports[`AllOrOne returns null nothing is passed in and showAll is true 1`] = `null`;

exports[`AllOrOne returns the first in a collection if showAll is false 1`] = `
<div>
  <p
    title="123 lol Street, Funmouth, MA 12345"
  >
    123 lol Street, Funmouth, MA 12345
     (
    11.41 mi
    )
  </p>
</div>
`;
